lib/travis/model/job/queue.rb

```diff
@@ -8,119 +8,81 @@ class Job
   # queue) or the language given in the configuration (`.travis.yml`) and
   # default to 'builds.linux'.
   class Queue
+    SUDO_AND_SETUID_EXECUTABLES = %w(
+      sudo
+      ping
+    )
+
+    SUDO_DETECTION_REGEXP = /\A[^#]*\b(#{SUDO_AND_SETUID_EXECUTABLES.join('|')})\b/
+
+    CUSTOM_STAGES = %w(
+      before_install
+      install
+      before_script
+      script
+      before_cache
+      after_success
+      after_failure
+      after_script
+    ).map(&:to_sym)
+
     class << self
       def for(job)
-        repo_name       = job.repository.try(:name)
-        owner_name      = job.repository.try(:owner_name)
-        language        = Array(job.config[:language]).flatten.compact.first
-        os              = job.config[:os]
-        sudo            = job.config[:sudo]
-        sudo_detected   = sudo_detected?(job.config)
-        owner           = job.repository.try(:owner)
-        education       = Travis::Github::Education.education_queue?(owner)
-        dist            = job.config[:dist]
-        repo_created_at = job.repository.created_at
-        queues.detect { |queue| queue.send(:matches?, owner_name, repo_name, language, os, sudo, education, dist, repo_created_at, sudo_detected) } || default
+        queues.detect(lambda { default }) { |queue| queue.matches?(job) }
       end
 
       def queues
         @queues ||= Array(Travis.config.queues).compact.map do |queue|
-          Queue.new(*queue.values_at(*[:queue, :slug, :owner, :language, :os, :sudo, :education, :dist]))
+          Queue.new(queue[:queue], queue.reject { |key, value| key == :queue })
         end
       end
 
       def default
-        @default ||= new(Travis.config.default_queue)
+        @default ||= Queue.new(Travis.config.default_queue, {})
       end
 
       def sudo_detected?(config)
-        config.values_at(*custom_stages).compact.map do |value|
-          Array(value).reject do |s|
-            s =~ /\s*#.*/
-          end.map do |s|
-            s =~ /\b(#{sudo_and_setuid_executables.join('|')})\b/
-          end.any?
-        end.any?
-      end
-
-      private
-
-        def sudo_and_setuid_executables
-          %w(
-            sudo
-            ping
-          )
+        config.values_at(*CUSTOM_STAGES).compact.flatten.any? do |s|
+           SUDO_DETECTION_REGEXP =~ s
```

Comment:
extra space thar

and :boom: I had no idea the operands of `=~` were interchangeable :stuck_out_tongue_winking_eye: